spec/operators/timeout-spec.js

```diff
@@ -1,44 +1,102 @@
-/* globals describe, it, expect */
+/* globals describe, it, expect, expectObservable, hot, rxTestScheduler */
 var Rx = require('../../dist/cjs/Rx');
 var Observable = Rx.Observable;
 
-describe('Observable.prototype.timeout', function () {
-  it('should timeout after a specified delay', function (done) {
-    Observable.never().timeout(100)
-      .subscribe(function (x) {
-        throw 'should not next';
-      }, function (err) {
-        expect(err.message).toBe('timeout');
-        done();
-      }, function () {
-        throw 'should not complete';
+describe('Observable.prototype.timeout()', function () {
+  var defaultTimeoutError = new Error('timeout');
+  
+  it('should timeout after a specified timeout period', function () {
+    var e1 = Observable.never();
+    var expected = '-----#';
+    
+    expectObservable(e1.timeout(50, null, rxTestScheduler)).toBe(expected, null, defaultTimeoutError);
+  });
+  
+  it('should timeout after specified timeout period and send the passed error', function () {
+    var e1 = Observable.never();
+    var expected = '-----#';
+    var value = 'hello';
+    
+    expectObservable(e1.timeout(50, value, rxTestScheduler)).toBe(expected, null, value);
+  });
+  
+  it('should timeout at a specified Date', function (done) {
+    var e1 = Observable.never();
+    
+    e1.timeout(new Date(Date.now() + 100), null)
```

Comment:
Gave a try, but not works by just replacing to testSchderuler's now(). This happens due to fixture setup using`Date`, 
1. `new Date(now + timeframe)` is delivered as absolute timeframe param
2. `timeout` calculate absolute time via `(+due - now)` where `due` is param above.

if use actual datetime it causes intermittent failure, and using testscheduler.now create 1 as negative timeframe since testscheduler.now returns 0 as starting timeframe.

I'd like to investigate this further as separate PR since it's test setup specific issue, may need few trial & errors.